core/modules/replica/MessengerConnector.cc

```diff
@@ -0,0 +1,640 @@
+        }
+    );
+    _id2request.erase(id);
+}
+
+bool MessengerConnector::exists(std::string const& id) const {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "exists  id=" << id);
+
+    util::Lock lock(_mtx, context() + "exists");
+
+    return _id2request.count(id);
+}
+
+void MessengerConnector::sendImpl(std::string const& id,
+                                  MessengerConnector::WrapperBase::Ptr const& ptr) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "sendImpl  id: " + id);
+
+    util::Lock lock(_mtx, context() + "sendImpl");
+
+    if (_id2request.count(id)) {
+        throw std::logic_error(
+                "MessengerConnector::sendImpl()  the request is alrady registered for id:" + id);
+    }
+
+    // Register the request
+
+    _id2request[id] = ptr;
+    _requests.push_back(ptr);
+
+    switch (_state) {
+
+        case STATE_INITIAL:
+            resolve(lock);
+            break;
+
+        case STATE_CONNECTING:
+            // Not ready to submit any requests before a connection
+            // is established.
+            break;
+
+        case STATE_COMMUNICATING:
+            sendRequest(lock);
+            break;
+    }
+}
+
+void MessengerConnector::restart(util::Lock const& lock) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "restart"
+         << "  _currentRequest->id=" << (_currentRequest ? _currentRequest->id : ""));
+
+    // Cancel any asynchronous operation(s) if not in the initial state
+
+    switch (_state) {
+
+        case STATE_INITIAL:
+            break;
+
+        case STATE_CONNECTING:
+        case STATE_COMMUNICATING:
+            _resolver.cancel();
+            _socket.cancel();
+            _socket.close();
+            _timer.cancel();
+
+            _state = STATE_INITIAL;
+            break;
+    }
+    resolve(lock);
+}
+
+void MessengerConnector::resolve(util::Lock const& lock) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "resolve"
+         << "  _currentRequest->id=" << (_currentRequest ? _currentRequest->id : ""));
+
+    if (_state != STATE_INITIAL) return;
+
+    boost::asio::ip::tcp::resolver::query query(
+        _workerInfo.svcHost,
+        std::to_string(_workerInfo.svcPort)
+    );
+    _resolver.async_resolve(
+        query,
+        boost::bind(
+            &MessengerConnector::resolved,
+            shared_from_this(),
+            boost::asio::placeholders::error,
+            boost::asio::placeholders::iterator
+        )
+    );
+    _state = STATE_CONNECTING;
+}
+
+void MessengerConnector::resolved(boost::system::error_code const& ec,
+                                  boost::asio::ip::tcp::resolver::iterator iter) {
+
+    LOGS(_log, LOG_LVL_DEBUG, context() << "resolved"
+         << "  _currentRequest->id=" << (_currentRequest ? _currentRequest->id : ""));
+
+    if (isAborted(ec)) return;
+
+    util::Lock lock(_mtx, context() + "resolved");
+
+    if (ec) waitBeforeRestart(lock);
+    else    connect(lock,
+                    iter);
```

Comment:
per the coding standard this need braces and expanding onto more lines.

Reply:
Fixed